packages/core/src/Engine.ts

```diff
@@ -106,6 +106,34 @@ export class Engine extends EventDispatcher {
     return this._settings;
   }
 
+  /**
+   * How this light casts shadows.
+   */
+  set shadowMode(value: ShadowMode) {
+    this._settings.shadowMode = value;
+  }
+
+  /**
+   * The default resolution of the shadow maps.
+   */
+  set shadowResolution(value: ShadowResolution) {
+    this._settings.shadowResolution = value;
+  }
+
+  /**
+   * Mode of cascades to use for directional light shadows.
+   */
+  set shadowCascades(value: ShadowCascadesMode) {
+    this._settings.shadowCascades = value;
+  }
+
+  /**
+   * The ratio of cascade distribution.
+   */
+  set shadowCascadeSplitRatio(value: number) {
+    this._settings.shadowCascadeSplitRatio = value;
+  }
```

Comment:
developer can use `engine.settings.shadowMode` to set this, later we will have more settings porperty, is they all in `Engine`,the API will messy.